frontend/src/components/accountPage/UserList/UserList.tsx

```diff
@@ -1,240 +1,204 @@
-        >
-          <PlusOutlined />
-        </Button>
-        <Upload
-          name="file"
-          beforeUpload={() => false}
-          accept=".csv"
-          onChange={onCsvUploaded}
-          fileList={[]}
-          maxCount={1}
+      </Row>
+      <Spin spinning={props.loading}>
+        <Table
+          pagination={{ defaultPageSize: 10 }}
+          dataSource={
+            searchText !== ''
+              ? props.users.filter(user => filterUser(user, searchText))
+              : props.users.sort(
+                  (u1: UserAccountPage, u2: UserAccountPage) =>
+                    u1.currentRole?.localeCompare(u2.currentRole!) || 0
+                )
+          }
+          size="small"
         >
-          <Button type="primary" className="m-1" icon={<UploadOutlined />}>
+          <Column
+            title="User ID"
+            dataIndex="userid"
+            sorter={(a: UserAccountPage, b: UserAccountPage) =>
+              a.userid.localeCompare(b.userid)
+            }
+            key="userid"
+            width={170}
+          />
+          <Column
+            responsive={['md', 'lg']}
+            title="Name"
+            dataIndex="name"
+            sorter={(a: UserAccountPage, b: UserAccountPage) =>
+              a.name.localeCompare(b.name)
+            }
+            key="name"
+            width={120}
+          />
+          <Column
+            responsive={['md', 'lg']}
+            title="Surname"
+            dataIndex="surname"
+            sorter={(a: UserAccountPage, b: UserAccountPage) =>
+              a.surname.localeCompare(b.surname)
+            }
+            key="surname"
+            width={120}
+          />
+          <Column
+            responsive={['sm', 'md', 'lg']}
+            title="Email"
+            dataIndex="email"
+            ellipsis={true}
+            key="email"
+            width={150}
+          />
+          <Column
+            responsive={['md', 'lg']}
+            title={<>Role</>}
+            dataIndex="currentRole"
+            sorter={(a: UserAccountPage, b: UserAccountPage) =>
+              a.currentRole?.localeCompare(b.currentRole!) || 0
+            }
+            key="currentRole"
+            width={80}
+          />
+          <Column
+            title="Action"
+            key="x"
+            width={60}
+            render={(_: any, record: UserAccountPage) =>
+              props.users.length >= 1 ? (
+                <div className="flex justify-center">
+                  <Tooltip title="Swap role">
+                    <SwapOutlined
+                      className="mr-2"
+                      onClick={() => handleChangeCurrentRole(record)}
+                    />
+                  </Tooltip>
+
+                  <Popconfirm
+                    title="Sure to delete?"
+                    onConfirm={async () => {
+                      props.onDeleteUser && (await props.onDeleteUser(record));
+                    }}
```

Comment:
```suggestion
                    onConfirm={async () => await props.onDeleteUser.?(record) }
```
Does this work?

Reply:
It does not work but I think the first `props.onDeleteUser &&` isn't needed since onDeleteUser function is not optional.
Same for onAddUser